### ISSUE TITLE
Perf: Structured parallel startup with futures::join!

### ISSUE THREAD

**Issue by boersmamarcel** [boersmamarcel/chatty2#203]
**Parent epic**: #196

## Problem

`main.rs:601-861` uses three `AtomicBool` flags (`providers_loaded`, `models_loaded`, `exec_settings_loaded`) to gate conversation loading. The loading tasks check these flags and only proceed when all three are true.

Current flow:
```
spawn(load_providers)     ──► sets providers_loaded = true
spawn(load_models)        ──► sets models_loaded = true  
spawn(load_exec_settings) ──► sets exec_settings_loaded = true
spawn(load_conversations) ──► polls all 3 flags, blocks until all true
```

The conversation loading task effectively busy-waits (or periodically checks) the atomic flags instead of using structured concurrency. This pattern:
- Is harder to reason about than `join!`
- Adds latency if the check interval isn't perfectly tuned
- Doesn't compose well if more dependencies are added

Additionally, 8 independent async tasks are spawned sequentially in main.rs:427-903. While they run concurrently via `cx.spawn()`, the spawn calls themselves are sequential, and there's no explicit grouping of independent vs. dependent loads.

## Goal

Replace atomic-bool polling with structured concurrency (`futures::join!` or `tokio::join!`). Make the dependency graph explicit and clear.

### Step 1: Group loads by dependency

Map the dependency graph:
```
Independent (no dependencies):
  - general_settings
  - user_secrets
  - training_settings
  - MCP servers
  - MCP listener

Tier 1 (independent but needed before conversations):
  - providers
  - models  
  - execution_settings

Tier 2 (depends on Tier 1):
  - conversations (needs providers + models + exec_settings)
```

### Step 2: Use join! for Tier 1

```rust
// Spawn Tier 1 loads and wait for all three
let (providers_result, models_result, exec_result) = futures::join!(
    load_providers(),
    load_models(),
    load_exec_settings(),
);

// Apply results to global state
cx.update(|cx| {
    cx.set_global(providers_result);
    cx.set_global(models_result);
    cx.set_global(exec_result);
}).ok();

// Now load conversations — dependencies guaranteed ready
load_conversations(cx).await;
```

### Step 3: Remove AtomicBool flags

Delete the `providers_loaded`, `models_loaded`, `exec_settings_loaded` AtomicBool flags. They're no longer needed because `join!` provides the synchronization.

### Step 4: Keep independent loads as fire-and-forget

General settings, user secrets, training settings, and MCP loads don't block anything — keep them as `cx.spawn(...).detach()`.

## Files to Modify

- `src/main.rs` — restructure startup sequence with `join!`, remove AtomicBool flags

## Acceptance Criteria

- [ ] AtomicBool polling flags are removed
- [ ] Providers, models, and exec_settings load in parallel via `join!`
- [ ] Conversations load only after all three dependencies are ready
- [ ] Independent loads (general settings, MCP, etc.) remain fire-and-forget
- [ ] Startup time is equal or faster (no regression)
- [ ] All existing tests pass

## Measurement

- Measure startup time with `std::time::Instant` before and after
- Expect marginal improvement (dependencies already ran in parallel), but the code is significantly clearer
